xdsl/parser.py

```diff
@@ -1429,7 +1327,8 @@ def try_parse_builtin_float_attr(self) -> FloatAttr | None:
                     "Float attribute must be typed with a float type!")
             return FloatAttr(float(value.text), type)
 
-    def try_parse_builtin_boolean_attr(self) -> IntegerAttr | None:
+    def try_parse_builtin_boolean_attr(
```

Comment:
Can builtin bool attr actually be of IndexType? IMO, we should only allow i1, which is an IntegerType, right?

Reply:
I don't mind. This one really isn't that big, but I see that this is somewhat out of scope. Your call...